src/pages/index.js

```diff
@@ -33,60 +47,165 @@ const cardSection = new Section({
+    })
+  })
+}
+
 function createCard(cardData){
-  const card = new Card(cardData, '.card-template', handleImageClick);
+  const card = new Card(cardData, '.card-template', handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick);
   return card.generateCard()
 };
 
 function handleProfileFormSubmit(data){
-  userInfo.setUserInfo(data);
+  this.renderLoading(true)
+  const {name, about} = data;
+  api.updateProfile(name, about).then(data => {
+    return data
+  }).then((info) => {
+    userInfo.setUserInfo(info)
+    formValidators['profileForm'].disableButton();
+    formValidators['profileForm'].resetInputs();
+    profileModal.close()
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    this.renderLoading(false)
+  })
 }
 
-// Instantiate Classes
-cardAddValidator.enableValidation()
-profileFormValidator.enableValidation()
-cardSection.renderItems();
+function handleLikeClick(id){
+  api.likeCard(id)
+  .then(()=>{
+    this.like()
+  })
+  .catch(err => {
+    console.error(err)
+  })
+}
+
+function handleDislikeClick(id){
+  api.dislikeCard(id)
+  .then(()=> {
+    this.dislike()
+  })
+  .catch(err => {
+    console.error(err)
+  })
+}
+
+function handleAvatarUpdate(info){
+  this.renderLoading(true)
+  api.updateAvatar(info.avatar).then((info)=>{
+    userInfo.setUserInfo(info)
+    formValidators['updateAvatar'].disableButton();
+    formValidators['updateAvatar'].resetInputs();
+    changeAvatarModal.close()
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    this.renderLoading(false)
+  })
+}
+
+const formValidators = {};
+function enableValidation(config){
+  const forms = Array.from(document.querySelectorAll(config.formSelector));
+  forms.forEach(form => {
+    const validator = new FormValidator(config, form);
+    const formName = form.getAttribute('id');
+    
+    formValidators[formName] = validator;
+    validator.enableValidation()
+  })
+}
+
+enableValidation(configuration)
+
+/* ------------------------------------------------------ */
+/*                   add event listeners                  */
+/* ------------------------------------------------------ */
 profileModal.setEventListeners();
 cardModal.setEventListeners();
 popupWithImage.setEventListeners()
-
-// add event listeners
+deleteConfirmModal.setEventListeners();
+changeAvatarModal.setEventListeners();
 editButton.addEventListener('click', ()=>{
   profileModal.open()
-  renderProfileDetails();
+  renderProfileDetails(profileModal);
 });
 
 addCardButton.addEventListener('click', ()=>{
   cardModal.open()
 });
+
+avatar.addEventListener('mouseover', ()=>{
+  avatarOverlay.classList.add('profile__photo-overlay_visible')
+})
+
+avatarOverlay.addEventListener('mouseout', ()=> {
+  avatarOverlay.classList.remove('profile__photo-overlay_visible')
+})
+
+avatarOverlay.addEventListener('click', ()=>{
+  changeAvatarModal.open()
+  formValidators['updateAvatar'].disableButton();
```

Comment:
You need to `disable` the button only after **resetting** the form when you have submitted a card/avatar otherwise users may close the modal by accident and they don’t expect that the button will be disabled on opening if the inputs have **valid** data which they haven’t submitted yet.